sklearn/preprocessing/data.py

```diff
@@ -1956,3 +1967,222 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class CountFeaturizer(BaseEstimator, TransformerMixin):
+
+    """
+    Adds in a new feature column containing the number of occurences of
+    each row in the given data set.
+
+    Formally, for each data point X_i in the dataset X, it will add in
+    a new feature count_X_i to the end of X_i where count_X_i is the number of
+    occurences of X_i in the dataset X given the equality indicator
+    'inclusion'
+
+    This preprocessing step is useful when the number of occurences
+    of a particular piece of data is helpful in computing the prediction.
+
+    Parameters
+    ----------
+    inclusion: set, list, numpy.ndarray, or string (only 'all'), default='all'
+        The inclusion criteria for counting
+        - 'all' (default): Every feature is taken into consideration when
+        counting how many of each row is in the dataset.
+        - collection of indices: A collection of values where if the index v is
+        in 'inclusion', then the vth feature will be put into the equality
+        checks when counting how often each row occurs.
+
+    removal_policy: set, list, numpy.ndarray, or None, default=None
```

Comment:
need space before colon.

I'd vote for `keep_original=True/False/indices`, however, I'd be tempted to simplify the interface: If a user wants to retain a set of features, they can use a `FeatureUnion`.